.github/workflows/openstack.yml

```diff
@@ -0,0 +1,301 @@
+              ./status_marker.sh 5
+              /scripts/elevate-cpanel --log &
+              REGEX="Rebooting into stage 5 of 5"
+              RETVAL=1
+              while [ $RETVAL -ne 0 ]; do grep "${REGEX}" /var/log/elevate-cpanel.log; RETVAL=$?; [ $RETVAL -eq 0 ] && echo "## [INFO] SUCCESS: Reboot text found in /var/log/elevate-cpanel.log  ##" && exit 0; RETRIES=$((RETRIES+1)); [ $RETVAL -ne 0 ] && echo "## [DEBUG]: Retrying Reboot REGEX Search: Attempt ${RETRIES} ..."; sleep 1; done
+
+    wait_for_stage_5_reboot:
+      runs-on: self-hosted
+      needs: watch_for_stage_5_reboot
+      outputs:
+        VM_IP: ${{ needs.watch_for_stage_5_reboot.outputs.VM_IP }}
+      steps:
+        - name: Wait For VM to Come Back From Stage 5 Reboot
+          working-directory: "./.github/workflows/openstack/"
+          run: |
+            ./ssh_retry.sh ${{ needs.watch_for_stage_5_reboot.outputs.VM_IP }}
+
+    watch_for_final_reboot:
+      runs-on: self-hosted
+      needs: wait_for_stage_5_reboot
+      outputs:
+        VM_IP: ${{ needs.wait_for_stage_5_reboot.outputs.VM_IP }}
+      steps:
+        - name: Watch Elevate for Final Reboot
+          uses: appleboy/ssh-action@v1.0.3
+          with:
+            host: ${{ needs.wait_for_stage_5_reboot.outputs.VM_IP }}
+            username: 'root'
+            key: ${{ secrets.SSH_PRIVATE_KEY }}
+            port: '22'
+            timeout: 30m
+            command_timeout: 20m
+            script: |
+              /scripts/elevate-cpanel --log &
+              REGEX="Doing final reboot"
+              RETVAL=1
+              while [ $RETVAL -ne 0 ]; do grep "${REGEX}" /var/log/elevate-cpanel.log; RETVAL=$?; [ $RETVAL -eq 0 ] && echo "## [INFO] SUCCESS: Reboot text found in /var/log/elevate-cpanel.log  ##" && exit 0; RETRIES=$((RETRIES+1)); [ $RETVAL -ne 0 ] && echo "## [DEBUG]: Retrying Reboot REGEX Search: Attempt ${RETRIES} ..."; sleep 1; done
+  
+    wait_for_final_reboot:
+      runs-on: self-hosted
+      needs: watch_for_final_reboot
+      outputs:
+        VM_IP: ${{ needs.watch_for_final_reboot.outputs.VM_IP }}
+      steps:
+        - name: Wait For VM to Come Back From Final Reboot
+          working-directory: "./.github/workflows/openstack/"
+          run: |
+            ./ssh_retry.sh ${{ needs.watch_for_final_reboot.outputs.VM_IP }}
+
+    verify_upgraded_os:
+      runs-on: self-hosted
+      needs: wait_for_final_reboot
+      outputs:
+        VM_IP: ${{ needs.wait_for_final_reboot.outputs.VM_IP }}
+      steps:
+        - name: Verify OS Upgraded and Elevate Complete
+          uses: appleboy/ssh-action@v1.0.3
+          with:
+            host: ${{ needs.wait_for_final_reboot.outputs.VM_IP }}
+            username: 'root'
+            key: ${{ secrets.SSH_PRIVATE_KEY }}
+            port: '22'
+            timeout: 5m
+            command_timeout: 1m
+            script: |
+              /usr/local/cpanel/3rdparty/bin/prove -lvm /opt/elevate/t/integration/*
```

Comment:
1. FIX:  This should be:  `/opt/elevate/t/integration/*.t`

2. QUESTION:  Do we think that there will be enough integration tests to warrant folders?  If so, then we may want to use find here to gather the tests:  `/usr/local/cpanel/3rdparty/bin/prove -lvm $(find /opt/elevate/t/integration/ -type f -name '*.t')`

3. CONSIDER:  Should we use `yath` here?

Reply:
I think, for now, t/integration/complete.t was the specific ask with plans to expand.  Expanding then won't be any complication so I think this linear approach works.

Yath def is my preference.

```/usr/local/cpanel/3rdparty/bin/yath test -v -j8 t/integration/*.t```